src/test/java/spoon/test/filters/FilterTest.java

```diff
@@ -466,4 +472,106 @@ public boolean matches(CtInvocation<?> element) {
+		launcher.run();
+		
+		class Context {
+			int counter = 0;
+		}
+		Context context = new Context();
+		
+		CtQuery<CtClass<?>> l_qv = launcher.getFactory().getModel().getRootPackage().filterChildren(new TypeFilter<>(CtClass.class));
+		
+		assertEquals(0, context.counter);
+		// map with java8 lambda
+		l_qv.map(cls->{
+			assertTrue(cls instanceof CtClass);
+			context.counter++;
+			return true;
+		}).list();
+		assertTrue(context.counter>0);
+	}
+	
+	@Test
+	public void testQueryBuilderWithFilterChain() throws Exception {
+		final Launcher launcher = new Launcher();
+		launcher.setArgs(new String[] {"--output-type", "nooutput" });
+		launcher.addInputResource("./src/test/java/spoon/test/filters/testclasses");
+		launcher.run();
+		
+		class Context {
+			CtMethod<?> method;
+			int count = 0;
+		}
+		
+		Context context = new Context();
+
+		launcher.getFactory().Package().getRootPackage().filterChildren(new TypeFilter<CtMethod<?>>(CtMethod.class))
+		// using a lazily-evaluated lambda
+		.map((CtMethod<?> method) -> {context.count++;return method;})
+		.list(); // actual evaluation
+		assertTrue(context.count>0);
+	}
+	
+	@Test
+	public void testFunctionQueryStep() throws Exception {
+		final Launcher launcher = new Launcher();
+		launcher.setArgs(new String[] {"--output-type", "nooutput","--level","info" });
+		launcher.addInputResource("./src/test/java/spoon/test/filters/testclasses");
+		launcher.run();
+		
+		class Context {
+			int count = 0;
+		}
+		
+		Context context = new Context();
+
+		CtQuery<CtType<?>> query = launcher.getFactory().Package().getRootPackage().filterChildren((CtClass<?> c)->{return true;})
+			.map((CtClass<?> c)->c.getSuperInterfaces())
+			.map((CtTypeReference<?> iface)->iface.getTypeDeclaration())
+			.map((CtType<?> iface)->iface.getAllMethods())
+			.map((CtMethod<?> method)->method.getSimpleName().equals("make"))
+			.map((CtMethod<?> m)->m.getType())
+			.map((CtTypeReference<?> t)->t.getTypeDeclaration());
+		((CtQueryImpl<?>)query).logging(true);
+		query.map((CtInterface<?> c)->{
+				assertEquals("ITostada", c.getSimpleName());
+				context.count++;
+				return true;
+			}).list();
+		assertTrue(context.count>0);
+	}
+	
+	@Test
+	public void testElementFunctionQueryStep() throws Exception {
+		final Launcher launcher = new Launcher();
+		launcher.setArgs(new String[] {"--output-type", "nooutput","--level","debug" });
+		launcher.addInputResource("./src/test/java/spoon/test/filters/testclasses");
+		launcher.run();
+		
+		class Context {
+			int count = 0;
+			int totalCount = 0;
+		}
+		
+		Context context = new Context();
+
+		launcher.getFactory().Class().get(FieldAccessFilterTacos.class).map((CtClass clazz)->clazz.getAllFields())
+			.map((CtFieldReference<?> fr)->{
+				if("myfield".equals(fr.getSimpleName())){
+					context.count++;
+				}
+				context.totalCount++;
+				return true;
+			}).list();
+		assertTrue(context.count==1);
+		assertTrue(context.totalCount>1);
+		
```

Comment:
You are right this test brings nothing new, The testFunctionQueryStep already tests the same. I will remove this test